dist-material/application.yml

```diff
@@ -83,180 +85,230 @@ core:
+      dataSource.prepStmtCacheSize: ${SW_STORAGE_METABASE_PREP_STMT_CACHE_SQL_SIZE:250}
+      dataSource.prepStmtCacheSqlLimit: ${SW_STORAGE_METABASE_PREP_STMT_CACHE_SQL_LIMIT:2048}
+      dataSource.useServerPrepStmts: ${SW_STORAGE_METABASE_USE_SERVER_PREP_STMTS:true}
+    metadataQueryMaxSize: ${SW_STORAGE_METABASE_QUERY_MAX_SIZE:5000}
+    # InfluxDB configuration
+    url: ${SW_STORAGE_INFLUXDB_URL:http://localhost:8086}
+    user: ${SW_STORAGE_INFLUXDB_USER:root}
+    password: ${SW_STORAGE_INFLUXDB_PASSWORD:}
+    database: ${SW_STORAGE_INFLUXDB_DATABASE:skywalking}
+    actions: ${SW_STORAGE_INFLUXDB_ACTIONS:1000} # the number of actions to collect
+    duration: ${SW_STORAGE_INFLUXDB_DURATION:1000} # the time to wait at most (milliseconds)
+    fetchTaskLogMaxSize: ${SW_STORAGE_INFLUXDB_FETCH_TASK_LOG_MAX_SIZE:5000} # the max number of fetch task log in a request
+
 receiver-sharing-server:
+  selector: ${SW_RECEIVER_SHARING_SERVER:default}
   default:
+    authentication: ${SW_AUTHENTICATION:""}
 receiver-register:
+  selector: ${SW_RECEIVER_REGISTER:default}
   default:
+
 receiver-trace:
+  selector: ${SW_RECEIVER_TRACE:default}
   default:
     bufferPath: ${SW_RECEIVER_BUFFER_PATH:../trace-buffer/}  # Path to trace buffer files, suggest to use absolute path
     bufferOffsetMaxFileSize: ${SW_RECEIVER_BUFFER_OFFSET_MAX_FILE_SIZE:100} # Unit is MB
     bufferDataMaxFileSize: ${SW_RECEIVER_BUFFER_DATA_MAX_FILE_SIZE:500} # Unit is MB
     bufferFileCleanWhenRestart: ${SW_RECEIVER_BUFFER_FILE_CLEAN_WHEN_RESTART:false}
     sampleRate: ${SW_TRACE_SAMPLE_RATE:10000} # The sample rate precision is 1/10000. 10000 means 100% sample in default.
     slowDBAccessThreshold: ${SW_SLOW_DB_THRESHOLD:default:200,mongodb:100} # The slow database access thresholds. Unit ms.
+
 receiver-jvm:
+  selector: ${SW_RECEIVER_JVM:default}
   default:
+
 receiver-clr:
+  selector: ${SW_RECEIVER_CLR:default}
   default:
+
 receiver-profile:
+  selector: ${SW_RECEIVER_PROFILE:default}
   default:
+
 service-mesh:
+  selector: ${SW_SERVICE_MESH:default}
   default:
     bufferPath: ${SW_SERVICE_MESH_BUFFER_PATH:../mesh-buffer/}  # Path to trace buffer files, suggest to use absolute path
     bufferOffsetMaxFileSize: ${SW_SERVICE_MESH_OFFSET_MAX_FILE_SIZE:100} # Unit is MB
     bufferDataMaxFileSize: ${SW_SERVICE_MESH_BUFFER_DATA_MAX_FILE_SIZE:500} # Unit is MB
     bufferFileCleanWhenRestart: ${SW_SERVICE_MESH_BUFFER_FILE_CLEAN_WHEN_RESTART:false}
+
 istio-telemetry:
+  selector: ${SW_ISTIO_TELEMETRY:default}
   default:
+
 envoy-metric:
+  selector: ${SW_ENVOY_METRIC:default}
   default:
-#    alsHTTPAnalysis: ${SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS:k8s-mesh}
-#receiver_zipkin:
-#  default:
-#    host: ${SW_RECEIVER_ZIPKIN_HOST:0.0.0.0}
-#    port: ${SW_RECEIVER_ZIPKIN_PORT:9411}
-#    contextPath: ${SW_RECEIVER_ZIPKIN_CONTEXT_PATH:/}
+    alsHTTPAnalysis: ${SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS:none}
+
+receiver_zipkin:
+  selector: ${SW_RECEIVER_ZIPKIN:none}
```

Comment:
What does this `none` mean? I don't think we have `none` provider for this receiver.

According to your implementation, https://github.com/apache/skywalking/blob/dae57dc8fddba8916d43351cdde824a116e2d58c/oap-server/server-bootstrap/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java#L62-L64,
this seems meaning no provider for module `receiver_zipkin`? If this module is required by other module, what happens?

I don't think providing a potential NPE in the runtime is a good idea.

Reply:
Also, based on this, I could put some tricky parameter value such as any not existing provider name. And we magically accept it and start up successfully.

Reply:
At the module core level, should be strict validation to avoid illegal selector name.

Reply:
> If this module is required by other module, what happens?
I don't think providing a potential NPE in the runtime is a good idea.
> Also, based on this, I could put some tricky parameter value such as any not existing provider name. And we magically accept it and start up successfully.

Can I just think of that this problem is not related to this pull request directly and it should be fixed in another pull request? In the old way, if the user just comment out the required module, what would happen? Are they the same problem here?

Reply:
Read old `ModuleDefine#prepare`. Once the provider doesn't exist, `throw new ProviderNotFoundException(this.name() + " module no provider exists.");` this shows up.

But because we enable multiple providers in YAML and using the selector, this check wouldn't work, that is why I asked.

Reply:
@wu-sheng hope I got your point correctly and I try to fix it at the phase when processing the configuration, to ensure that the processed configurations are the same as before, and verify like this, correct me if I made anything wrong still

<img width="1197" alt="image" src="https://user-images.githubusercontent.com/15965696/76698419-a6c5ae00-66dd-11ea-843f-ab26f112f444.png">

Reply:
This fails because some module request cluster module in the bootstrap stage, if not, then this error wouldn't show up. My point is, this should be a check rule.